### ISSUE TITLE
Training error in frcnn_anchor_target_layer.cpp:259

### ISSUE THREAD

**Issue by mychina75** [D-X-Y/caffe-faster-rcnn#16]
Hi, 
I composed dataset based on your method. After few iterations, error pop up and interupt training.
looks like does not found positive samples.

I0420 19:06:36.288517 32748 solver.cpp:225] Iteration 11760 (1.36487 iter/s, 14.6534s/20 iters), loss = 0.364464  [ 11760 / 350000 ] -> [ 6:25 (H:M) ]
I0420 19:06:36.288591 32748 solver.cpp:244]     Train net output #0: bbox_loss = 0.247996 (* 1 = 0.247996 loss)
I0420 19:06:36.288606 32748 solver.cpp:244]     Train net output #1: cls_loss = 0.425677 (* 1 = 0.425677 loss)
I0420 19:06:36.288619 32748 solver.cpp:244]     Train net output #2: rpn_cls_loss = 0.243714 (* 1 = 0.243714 loss)
I0420 19:06:36.288632 32748 solver.cpp:244]     Train net output #3: rpn_loss_bbox = 0.0200156 (* 1 = 0.0200156 loss)
I0420 19:06:36.288650 32748 sgd_solver.cpp:105] Iteration 11760, lr = 0.001
F0420 19:06:42.476500 32748 frcnn_anchor_target_layer.cpp:259] Check failed: num_examples > 0 (0 vs. 0) 
*** Check failure stack trace: ***
    @     0x7f819341edaa  (unknown)
    @     0x7f819341ece4  (unknown)
    @     0x7f819341e6e6  (unknown)
    @     0x7f8193421687  (unknown)
    @     0x7f8193bfddc3  caffe::Frcnn::FrcnnAnchorTargetLayer<>::Forward_cpu()
    @     0x7f8193b92541  caffe::Net<>::ForwardFromTo()
    @     0x7f8193b92637  caffe::Net<>::Forward()
    @     0x7f8193c4fef0  caffe::Solver<>::Step()
    @     0x7f8193c50fca  caffe::Solver<>::Solve()
    @           0x40bc19  train()
    @           0x408e53  main
    @     0x7f819240dec5  (unknown)
    @           0x4097b6  (unknown)
    @              (nil)  (unknown)

**Comment by D-X-Y** [comment]
You should check your dataset, the error log shows that in that image, the rpn cannot find any positive instances, maybe that image do not have any ground truth boxes.
